sklearn/cluster/_optics.py

```diff
@@ -710,6 +710,17 @@ def cluster_optics_dbscan(*, reachability, core_distances, ordering, eps):
     labels[far_reach & ~near_core] = -1
     return labels
 
+@validate_params(
+    {
+        "reachability": [np.ndarray],
+        "predecessor": [np.ndarray],
+        "ordering": [np.ndarray],
+        "min_samples":  [Interval(Integral, 1, None, closed="left"), Interval(Real, 1, 0, closed="both")],
+        "min_cluster_size":  [Interval(Integral, 1, None, closed="left"), Interval(Real, 1, 0, closed="both"),None],
+        "xi": [Interval(Real, 1, 0, closed="both"), 0.05],
+        "predecessor_correction": ["boolean"]
+    }
+)
 
```

Comment:
```suggestion

@validate_params(
    {
        "reachability": [np.ndarray],
        "predecessor": [np.ndarray],
        "ordering": [np.ndarray],
        "min_samples": [
            Interval(Integral, 1, None, closed="neither"),
            Interval(Real, 0, 1, closed="neither")
        ],
        "min_cluster_size": [
            Interval(Integral, 1, None, closed="neither"),
            Interval(Real, 0, 1, closed="neither"),
            None,
        ],
        "xi": [Interval(Real, 0, 1, closed="neither")],
        "predecessor_correction": ["boolean"]
    }
)
```

Reply:
Here are the necessary changes:

- follow PEP8 (and using `black`) as mentioned in the contribution guideline
- the docstring state that `min_samples` and `min_cluster_size` should be strictly greater than 1. We can also double check with the code that do `max(2, ...)`. Therefore, we need to set `closed=neither` for the `Interval` with `Integral` numbers.
- Reject 0 and 1 for the Real case.